tensorrt_llm/_torch/auto_deploy/transform/library/gather_logits_before_lm_head.py

```
@@ -0,0 +1,80 @@
+    Benefits:
+    - Reduces computation by only computing logits for needed tokens
+    - Eliminates Python loop overhead
+    - Enables CUDA graph capture of the gather
+    - Moves gather into the graph for better optimization
+    """
+
+    def _apply(
+        self,
+        gm: GraphModule,
+        cm,
+        factory,
```

Ruff (0.14.8)

Unused method argument: `factory`

(ARG002)